polkadot/zombienet-sdk-tests/tests/functional/approved_peer_mixed_validators.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Test what happens with a mixed fleet of validators (new ones which allow the ApprovedPeer
+// UMPSignal and old ones which consider these invalid during backing). Collators that don't send
+// these signals should not be affected, while collators sending the signal will see degraded
+// backing throughput. No disputes should be raised and finality is not affected.
+
+use anyhow::anyhow;
+
+use cumulus_zombienet_sdk_helpers::{assert_finality_lag, assert_finalized_para_throughput};
+use polkadot_primitives::Id as ParaId;
+use serde_json::json;
+use subxt::{OnlineClient, PolkadotConfig};
+use zombienet_sdk::NetworkConfigBuilder;
+
+#[tokio::test(flavor = "multi_thread")]
+async fn approved_peer_mixed_validators_test() -> Result<(), anyhow::Error> {
+	let _ = env_logger::try_init_from_env(
+		env_logger::Env::default().filter_or(env_logger::DEFAULT_FILTER_ENV, "info"),
+	);
+
+	let images = zombienet_sdk::environment::get_images_from_env();
+
+	let config = NetworkConfigBuilder::new()
+		.with_relaychain(|r| {
+			let r = r
+				.with_chain("rococo-local")
+				.with_default_command("polkadot")
+				.with_default_image(images.polkadot.as_str())
+				.with_default_args(vec![("-lparachain=debug").into()])
+				.with_genesis_overrides(json!({
+					"configuration": {
+						"config": {
+							"relay_vrf_modulo_samples": 2,
```

Comment:
@alexggh since this test has 2 cores, is this the right setting to ensure that all validators perform approval-checking on all candidates?